TestsFoundation/TestsFoundation/Element/Element.swift

```diff
@@ -127,32 +134,29 @@ public struct XCUIElementWrapper: Element {
         return element.exists
     }
     public var frame: CGRect {
-        guard exists else { return .zero }
         return element.frame
     }
     public var id: String {
-        guard exists else { return "" }
         return element.identifier
     }
     public var isEnabled: Bool {
         guard exists else { return false }
         return element.isEnabled
     }
     public var isSelected: Bool {
-        guard exists else { return false }
         return element.isSelected
     }
     public var isVisible: Bool {
         guard exists else { return false }
         return element.isHittable
     }
-    public var label: String {
-        guard exists else { return "" }
+    public func label(file: StaticString = #file, line: UInt = #line) -> String {
+        waitToExist(30, file: file, line: line)
         return element.label
     }
-    public var value: String {
-        guard exists else { return "" }
-        return element.value as? String ?? ""
+    public func value(file: StaticString = #file, line: UInt = #line) -> String? {
+        waitToExist(30, file: file, line: line)
+        return element.value as? String
```

Comment:
our pattern had been to not use waitToExist in the attribute getters, but I guess since we were checking exists (which is all that happens in the fast path of waitToExist) it's probably better like this.

Reply:
This fixes a lot of flakiness. Often the wrong thing, or not enough things were being waited on, causing race conditions that only showed up when tests were running quickly.

Reply:
👍